### ISSUE TITLE
Create TDS Fees (Ready for PROD)

### ISSUE THREAD

**Issue by TracyLinder** [cityofaustin/atd-data-tech#7017]
TDS Fee Spreadsheet is [here](https://docs.google.com/spreadsheets/d/17Wso9esPgv1deUspTYz-v_VyzZxDeL_10uGaXe_8YYg/edit?usp=sharing)

**Acceptance Criteria**
The fees provided below need to be configured/updated as specified:

**DAC Fees (Residential/Sidewalk Waivers) Review**
- Fee code: New Fee that needs to be configured
- Fee Description: ATD Residential Sidewalk Waiver Review 
- Fee Amount: $35
- Fee Insertion Trigger: 
-- Fee should be inserted when user manually clicks on Add -> Insert and Run Procedure in Fee Tab for PR folders. Fee does not need to be inserted in any other scenario.
-- If there are validations in PR folders to stop users from proceeding due to unpaid fees, these fees do not need to be a part of that validation. (TBD)
-- UAT: We need DSD to help / UAT  👆🏼

**Site Plan Corrections**
- Fee Code: New Fee that needs to be configured
- Fee Description: ATD - Site Plan Correction Review
- Fee amount: $75
- Fee Insertion Trigger: 
-- Fees need to be inserted when Accepted attempt is made in Correction Intake Process in SC Folder. 
-- DSD SPOC: Victor Auzenne

**Transportation Completeness Check**
- Fee Code: New Fee that needs to be configured
- Fee Description: ATD - Transportation Completeness Check
- Fee amount: $25
- Fee Insertion Trigger: 
-- Fees should be inserted when 'Calculate fees' attempt is made on 'Initial Intake' process (51110) in C8, C and SP Folders

**PDA Amendment** 
- Fee Code: New Fee that needs to be configured
- Fee Description: ATD - Planned Development Agreement
- Fee amount: $2810
- Fee Insertion Trigger: 
-- Fees should be inserted when 'Calculate fees' attempt is made on 'Submittal Intake' process (51120) in ZC Folders. 
-- PAZ SPOC is Victor Cortez

**ATD Tech Surcharge**
- Fee Description: ATD Tech Surcharge TDS
- Fee Amount: 4% of any of the above fees inserted in the folder. 
- Fee Insertion Trigger: Fees should be inserted when any of the fees mentioned above are inserted in the folder.

**Comment by TracyLinder** [comment]
Tracy will schedule sync w/ our Devs early next week to discuss this one.

**Comment by TracyLinder** [comment]
Note to us: We need to make a new fee for the TDS surcharge so we can report on it separately from the ROW surcharge.

**Comment by vschandramouli** [comment]
@TracyLinder I updated the AC to say the Tech Surcharge will be a new fee. Do you want to name it something else so there is no confusion between this and ROW Surcharge?

**Comment by TracyLinder** [comment]
Questions on this one:
1. When will development begin?
2. How long will it take?

**Comment by TracyLinder** [comment]
### Update

Priorities for TDS have been updated. Stakeholder wants us to deliver #4989 before we deliver this one. @Lpriya-LIT please finish your analysis and add any notes that might be useful when we start working on this in a few weeks. We will assign another card to you. FYI @Nadin-Nader

**Comment by Lpriya-LIT** [comment]
Notes : 
1. Add 4 new fee codes
2. Need GL code
3. Flat fee ( Need to add fee amount in lookup table)
4. Coding to insert fee based on the fee triggers mentioned in the description
5. Confirm whether the fee has to be auto billed.

**Comment by TracyLinder** [comment]
This is about a week's worth of Dev time.

**Comment by TracyLinder** [comment]
### Update

@Lpriya-LIT please move forward with development.

**Comment by TracyLinder** [comment]
### GL Codes

From Van:


DAC fees 5125-2400-9100-4163;

Site Plan Corrections  5125-2400-9100-4163;

Transportation completeness Check  5125-2400-9100-4163;

PUD Amendment  5125-2400-9100-4163;

Technology Improvement Surcharge 5125-2400-9301-4187.

**Comment by TracyLinder** [comment]
@vschandramouli can you please update AC. Each fee needs to insert w/ the ATD Tech Surcharge TDS.

**Comment by Lpriya-LIT** [comment]
Update - Surcharge fee name has been updated as " ATD Tech Surcharge TDS "

**Comment by TracyLinder** [comment]
**Status:** Still in development in Local

**Comment by Nadin-Nader** [comment]
**Status:**
LakshmiPriya has done the coding for this on Local but is having connectivity issues and cannot TEST and commit the changes yet. She will reach out to Martin for help if by 4pm 11/3/21 the issue is not resolved.

Trying to get this on TEST for Stakeholder UAT on Friday or earlier

**Comment by Nadin-Nader** [comment]
**Status:** George is currently working on deploying the NEW changes to DEV

**Comment by Lpriya-LIT** [comment]
@Nadin-Nader Changes deployed to Dev. Verified and it looks good.
It is ready for Test Promotion

**Comment by Nadin-Nader** [comment]
@Lpriya-LIT this has been sent to the DevOps CM Team to deploy to **TEST** under **_SCTASK0179633_**

**Comment by Nadin-Nader** [comment]
**Status:**
Waiting on the DevOps CM Team to deploy to **TEST** under **_SCTASK0179633_**

**Comment by TracyLinder** [comment]
@vschandramouli will facilitate UAT with Sangeeta next week.

**Comment by Nadin-Nader** [comment]
@Lpriya-LIT @vschandramouli @TracyLinder this has been deployed to **TEST** and s ready for UAT with Stakeholders

**Comment by vschandramouli** [comment]
11/22 - Completed UAT with Stakeholders (Sangeeta), all fees are inserted correctly, this item can be moved to PROD. 

CC @Nadin-Nader  @TracyLinder  @Lpriya-LIT

**Comment by Nadin-Nader** [comment]
@Lpriya-LIT Please go ahead and get the deployment plan to me so we can get these on PROD

**Comment by Nadin-Nader** [comment]
@Lpriya-LIT I have sent this over to the DevOps CM Team to be deployed to **_PROD_** under **SCTASK0182351**

**Comment by Nadin-Nader** [comment]
@Lpriya-LIT @vschandramouli @TracyLinder This has been deployed to **PROD**. I will send an email to Sangeeta and close the card.

**Comment by Nadin-Nader** [comment]
Email to Sangeeta:

> Hi Sangeeta,
> 
> Hope you are having a wonderful day and Holiday. I was emailing to inform you that the TDS Fees are now on PROD per your approval. 
> 
> Please let me know if you have questions or concerns. I will go ahead and close the card. If anything please feel free to reach out.
> 
> Respectfully,
> 
> Nadin A. Nader (He/Him/His)
> Product Delivery Manager
> Data and Technology Services
>
